security/pkg/nodeagent/caclient/providers/google/client.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package caclient
+
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+
+	"istio.io/istio/pkg/log"
+	caClientInterface "istio.io/istio/security/pkg/nodeagent/caclient/interface"
+	gcapb "istio.io/istio/security/proto/providers/google"
```

Comment:
I don't think we need to differentiate the proto for Google. Citadel will align with Google CA. We need to merge the following two protos:
https://github.com/istio/istio/blob/master/security/proto/istioca.proto
https://github.com/istio/istio/blob/master/security/proto/providers/google/istioca.proto

What do you think?

Reply:
Since different CA proto is defined in different pkg in server side, different ca provider need to add their own proto other it hit the error like below
```
Failed to create certificate: rpc error: code = Unimplemented desc = The GRPC(loas2) target is not implemented on the server, host: istioca.googleapis.com, method: /istio.v1.auth.IstioCertificateService/CreateCertificate.
```